internal/versions/common.go

```diff
@@ -160,9 +160,15 @@ func validateReleaseImageForRHCOS(
 				}
 			}
 		}
+		if swag.ContainsStrings(releaseImage.CPUArchitectures, cpuArchitecture) && !swag.ContainsStrings(availableOCPVersions, swag.StringValue(minorVersion)) {
+			availableOCPVersions = append(availableOCPVersions, swag.StringValue(releaseImage.OpenshiftVersion))
+		}
 	}
-
-	return errors.Errorf("The requested RHCOS version (%s, arch: %s) does not have a matching OpenShift release image", *rhcosVersionPtr, cpuArchitecture)
+	ocpVersionAdvice := fmt.Sprintf("There are no OCP versions available in release images for arch %s.", cpuArchitecture)
```

Comment:
This function validates that there is a OCP release image that matches the given RHCOS version/arch, I don't think its job is to give the user advice. I would give the advice probably form `RegisterInfraEnvInternal` or `validateArchitectureAndVersion`

Reply:
I disagree..I think that the knowledge of whether we have a matching image comes from this area and we only care to advise the user when there is no release image.

Reply:
I actually disagree.

It's true that `validateInfraEnvCreateParams` makes a call to `validateArchitectureAndVersion`, so you could provide some mechanism through which you provide "advice" when you receive an error from this path. 

It should however be noted that we don't know the real cause of the error as `validateArchitectureAndVersion` can return an error in a couple of scenarios.

To assume that we should provide advice on any error is incorrect.

It is in fact simpler to add the advice here where we have full access to all the facts and we waste no further CPU cycles obtaining it.

Reply:
My argument is that this function can potentially be used outside of the infraenv scope, which therefore will be wrong to display such an error. why not add some error matching mechanism one or two levels up the call stack, where the context is clear ?

Reply:
I guess that at the infraenv scope, we could have an explicit validation to enforce this?

I think we would still have to just create a redundant validation as error matching on the text of the error message? That seems to be a bit of a bad practice surely?

Realistically, would this be called outside of the infraenv scope? Does that happen already?